services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -626,6 +574,107 @@ def get_writer_batch_size(ds_config_info: datasets.info.DatasetInfo) -> Optional
+            src_revision, src_path_in_repo = data_file.split("/datasets/" + builder.repo_id + "/resolve/", 1)[1].split(
+                "/", 1
+            )
+
+            # for forward compatibility with https://github.com/huggingface/datasets/pull/5331
+            parquet_name = str(builder.dataset_name) if hasattr(builder, "dataset_name") else builder.name
+
+            if num_shards > 1:
+                path_in_repo = (
+                    f"{builder.config.name}/{parquet_name}-{split}-{shard_idx:05d}-of-{num_shards:05d}.parquet"
+                )
+            else:
+                path_in_repo = f"{builder.config.name}/{parquet_name}-{split}.parquet"
+
+            parquet_operations.append(
+                CommitOperationCopy(
+                    src_path_in_repo=src_path_in_repo, path_in_repo=path_in_repo, src_revision=src_revision
+                )
+            )
+    return parquet_operations
+
+
+def get_parquet_file_and_size(url: str, fs: HTTPFileSystem, hf_token: Optional[str]) -> Tuple[pq.ParquetFile, int]:
+    headers = get_authentication_headers_for_url(url, use_auth_token=hf_token)
+    f = fs.open(url, headers=headers)
+    return pq.ParquetFile(f), f.size
+
+
+def fill_builder_info(builder: DatasetBuilder, hf_token: Optional[str]) -> None:
+    """Fill the builder DatasetInfo from the copeid parquet files"""
+    data_files = builder.config.data_files
+    if not data_files:
+        raise EmptyDatasetError("Empty parquet data_files")
+    fs = HTTPFileSystem()
+    if not builder.info.splits or not builder.info.download_size:
+        builder.info.splits = SplitDict()
+        builder.info.dataset_size = 0
+        for split in data_files:
+            try:
+                parquet_files_and_sizes: List[Tuple[pq.ParquetFile, int]] = thread_map(
+                    partial(get_parquet_file_and_size, fs=fs, hf_token=hf_token),
+                    data_files[split],
+                    unit="pq",
+                    disable=True,
+                )
+                parquet_files, sizes = zip(*parquet_files_and_sizes)
+            except Exception as e:
+                raise FileSystemError(f"Could not read the parquet files: {e}") from e
+            if parquet_files:
+                first_pf = parquet_files[0]
+                if builder.info.features is None:
+                    builder.info.features = Features.from_arrow_schema(first_pf.schema_arrow)
+                first_row_group = first_pf.read_row_group(0)
+                compression_ratio = first_row_group.nbytes / first_row_group.num_rows
+                num_examples = sum(parquet_file.metadata.num_rows for parquet_file in parquet_files)
+                approx_num_bytes = int(compression_ratio * num_examples)
+                builder.info.splits.add(SplitInfo(split, num_bytes=approx_num_bytes, num_examples=num_examples))
+                builder.info.download_size += sum(sizes)
+                builder.info.dataset_size += approx_num_bytes
```

Comment:
OK! Anyway: is it a useful information anyway for the Hub users? I don't think so, maybe we should even hide it

Reply:
as you want - but I'd keep the approximate info here anyway